src/test/java/net/snowflake/client/jdbc/BindingDataLatestIT.java

```diff
@@ -60,57 +64,67 @@ public void testBindTimestampTZ() throws SQLException {
-      statement.execute("alter session set CLIENT_TIMESTAMP_TYPE_MAPPING=TIMESTAMP_NTZ");
-      statement.execute("alter session set TIMEZONE='Asia/Tokyo'");
-      TimeZone.setDefault(TimeZone.getTimeZone("Asia/Tokyo"));
-      Timestamp currT = new Timestamp(System.currentTimeMillis());
-
-      // insert using stage binding
-      PreparedStatement prepStatement =
-          connection.prepareStatement("insert into stageinsert values (?,?,?,?)");
-      statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1");
-      prepStatement.setInt(1, 1);
-      prepStatement.setTimestamp(2, currT);
-      prepStatement.setTimestamp(3, currT);
-      prepStatement.setTimestamp(4, currT);
-      prepStatement.addBatch();
-      prepStatement.executeBatch();
-      statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 0");
-
-      // insert using regular binging
-      prepStatement = connection.prepareStatement("insert into regularinsert values (?,?,?,?)");
-      for (int i = 1; i <= 6; i++) {
-        prepStatement.setInt(1, 1);
-        prepStatement.setTimestamp(2, currT);
-        prepStatement.setTimestamp(3, currT);
-        prepStatement.setTimestamp(4, currT);
-        prepStatement.addBatch();
+    TimeZone origTz = TimeZone.getDefault();
+    Calendar utc = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+    TimeZone.setDefault(TimeZone.getTimeZone("Asia/Tokyo"));
+
+    try (Connection connection = getConnection();
+        Statement statement = connection.createStatement()) {
+      try {
+        statement.execute(
+            "create or replace table stageinsert(ind int, ltz0 timestamp_ltz, tz0 timestamp_tz, ntz0 timestamp_ntz)");
+        statement.execute(
+            "create or replace table regularinsert(ind int, ltz0 timestamp_ltz, tz0 timestamp_tz, ntz0 timestamp_ntz)");
+        statement.execute("alter session set CLIENT_TIMESTAMP_TYPE_MAPPING=TIMESTAMP_NTZ");
+        statement.execute("alter session set TIMEZONE='Asia/Tokyo'");
+        Timestamp currT = new Timestamp(System.currentTimeMillis());
+
+        // insert using regular binging
+        try (PreparedStatement prepStatement =
+            connection.prepareStatement("insert into regularinsert values (?,?,?,?)")) {
+          for (int i = 1; i <= 6; i++) {
+            prepStatement.setInt(1, 1);
+            prepStatement.setTimestamp(2, currT);
+            prepStatement.setTimestamp(3, currT);
+            prepStatement.setTimestamp(4, currT);
+            prepStatement.addBatch();
+          }
+          prepStatement.executeBatch();
+        }
+
+        // insert using stage binding
+        try (PreparedStatement prepStatement =
+            connection.prepareStatement("insert into stageinsert values (?,?,?,?)")) {
+          statement.execute("ALTER SESSION SET CLIENT_STAGE_ARRAY_BINDING_THRESHOLD = 1");
```

Comment:
let's move altering session before try and create a method to execute prepare statement and do not duplicate the code

Reply:
Done

Reply:
FYI: About creating a method to execute preparestatment, I made a function. However, it cannot cover all the preparestatement in the testings because some of them need to pass the timezone as a parameter for the conversion when the preparestatement executes 'setTimestamp'. I used the method where the preprestatement does not need the timezone. Please have a look.

Reply:
It's fine - let's reuse as much possible but not too much